bftengine/src/bftengine/ReadOnlyReplica.cpp

```diff
@@ -83,6 +88,10 @@ void ReadOnlyReplica::sendAskForCheckpointMsg() {
 
 template <>
 void ReadOnlyReplica::onMessage<CheckpointMsg>(CheckpointMsg *msg) {
+  if (isCollectingState()) {
```

Comment:
Could you please explain the rationale for this change? I assume it is not related to this PR, but I might be wrong here.

Reply:
Previously we prevented any message from being processed while ST. now we want to allow client request and deny any other message (Checkpoint message)

Reply:
Thanks!